pkg/kubelet/status_manager.go

```diff
@@ -151,3 +168,14 @@ func (s *statusManager) syncBatch() error {
 	s.DeletePodStatus(podFullName)
 	return fmt.Errorf("error updating status for pod %q: %v", pod.Name, err)
 }
+
+// notRunning returns true if every status is terminated or waiting, or the status list
+// is empty.
+func notRunning(statuses []api.ContainerStatus) bool {
+	for _, status := range statuses {
+		if status.State.Terminated == nil && status.State.Waiting == nil {
```

Comment:
How about changing the function to something like "stillRunning", and replace this with `if status.State.Running != nil { return true}` ?